### PR TITLE
Add kube-aws 0.8.1

### PR DESCRIPTION
**Pull request by jskswamy**
- [✓] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [✓] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [✓] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [✓] Does your submission pass `brew audit --new-formula <formula>` (after doing `brew install <formula>`)?

---

### CODE REVIEW

File: Formula/kube-aws.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  desc "Kubernetes on AWS"
+  homepage "https://github.com/coreos/coreos-kubernetes/tree/master/multi-node/aws"
+  url "https://github.com/coreos/coreos-kubernetes/archive/v0.8.1.tar.gz"
+  sha256 "fe65c3fb72af7aa47dad07b8428eace077314edd3c129dba1fea5d3cccfaf094"
+  head "https://github.com/coreos/coreos-kubernetes.git"
+
+  depends_on "go" => :build
+
+  def install
+    executable="bin/kube-aws"
```
Comment (dunn) [inline Formula/kube-aws.rb]: Could you add spaces on each side of operators? It makes things a lot more readable.

File: Formula/kube-aws.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    ENV["GOPATH_VENDOR"]="#{Dir.pwd}/_gopath-vendor"
+    ENV["GOPATH_KUBE_AWS"]="#{Dir.pwd}/_gopath-kube-aws"
+    ENV["KUBE_AWS_DIR"]="#{ENV["GOPATH_KUBE_AWS"]}/src/github.com/coreos/coreos-kubernetes/multi-node/aws"
+    ENV["OUTPUT_PATH"]=executable.to_s
+    ENV["GOPATH"]="#{ENV["GOPATH_VENDOR"]}:#{ENV["GOPATH_KUBE_AWS"]}"
+
+    Dir.chdir "#{Dir.pwd}/multi-node/aws/"
+
+    rm_rf %w(ENV['GOPATH_VENDOR'] ENV['GOPATH_KUBE_AWS'])
```
Comment (dunn) [inline Formula/kube-aws.rb]: I think you want `[ENV['GOPATH_VENDOR'], ENV['GOPATH_KUBE_AWS']]` here.

File: Formula/kube-aws.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+    mkdir_p ENV["GOPATH_VENDOR"]
+    mkdir_p File.dirname(ENV["KUBE_AWS_DIR"])
+
+    ln "#{Dir.pwd}/vendor", "#{ENV["GOPATH_VENDOR"]}/src"
+    ln Dir.pwd.to_s, (ENV["KUBE_AWS_DIR"]).to_s
+
+    system "go", "generate", "./pkg/config"
+    system "go build -ldflags '-X github.com/coreos/coreos-kubernetes/multi-node/aws/pkg/cluster.VERSION=#{version}' -a -tags netgo -installsuffix netgo -o '#{ENV["OUTPUT_PATH"]}' ./cmd/kube-aws"
```
Comment (dunn) [inline Formula/kube-aws.rb]: Pass each part of the string as a separate argument, like
```ruby
system "go", "build", "-ldflags" ...

Comment (dunn) [inline Formula/kube-aws.rb]: And instead of `ENV['OUTPUT_PATH']`, you can just build it directly to `bin`.

File: Formula/kube-aws.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+    bin.install "#{Dir.pwd}/#{executable}" => "kube-aws"
+  end
+
+  test do
+    output = shell_output("#{bin}/kube-aws version 2>&1")
+    assert_match "kube-aws version #{version}", output
```
Comment (dunn) [inline Formula/kube-aws.rb]: It'd be great to have a test that exercises the functionality more.

File: Formula/kube-aws.rb [hunk 5]
```diff
@@ -8,16 +10,15 @@ class KubeAws < Formula
   depends_on "go" => :build
 
   def install
-    executable="bin/kube-aws"
+    executable = "bin/kube-aws"
```
Comment (jskswamy) [inline Formula/kube-aws.rb]: Added space between operator for better readability

File: Formula/kube-aws.rb [hunk 6]
```diff
@@ -8,16 +10,15 @@ class KubeAws < Formula
...
     ENV["GOPATH_VENDOR"]="#{Dir.pwd}/_gopath-vendor"
     ENV["GOPATH_KUBE_AWS"]="#{Dir.pwd}/_gopath-kube-aws"
     ENV["KUBE_AWS_DIR"]="#{ENV["GOPATH_KUBE_AWS"]}/src/github.com/coreos/coreos-kubernetes/multi-node/aws"
-    ENV["OUTPUT_PATH"]=executable.to_s
     ENV["GOPATH"]="#{ENV["GOPATH_VENDOR"]}:#{ENV["GOPATH_KUBE_AWS"]}"
 
     Dir.chdir "#{Dir.pwd}/multi-node/aws/"
 
-    rm_rf %w(ENV['GOPATH_VENDOR'] ENV['GOPATH_KUBE_AWS'])
+    rm_rf [ENV["GOPATH_VENDOR"], ENV["GOPATH_KUBE_AWS"]]
```
Comment (jskswamy) [inline Formula/kube-aws.rb]: Prefer array syntax over %w syntax

File: Formula/kube-aws.rb [hunk 7]
```diff
@@ -26,13 +27,18 @@ def install
     ln Dir.pwd.to_s, (ENV["KUBE_AWS_DIR"]).to_s
 
     system "go", "generate", "./pkg/config"
-    system "go build -ldflags '-X github.com/coreos/coreos-kubernetes/multi-node/aws/pkg/cluster.VERSION=#{version}' -a -tags netgo -installsuffix netgo -o '#{ENV["OUTPUT_PATH"]}' ./cmd/kube-aws"
+    system "go", "build", "-ldflags", "-X github.com/coreos/coreos-kubernetes/multi-node/aws/pkg/cluster.VERSION=#{version}", "-a", "-tags", "netgo", "-installsuffix", "netgo", "-o", executable.to_s, "./cmd/kube-aws"
```
Comment (jskswamy) [inline Formula/kube-aws.rb]: Prefer multiple arguments over single argument

File: Formula/kube-aws.rb [hunk 8]
```diff
@@ -26,13 +27,18 @@ def install
...
 
   test do
-    output = shell_output("#{bin}/kube-aws version 2>&1")
-    assert_match "kube-aws version #{version}", output
+    cluster = { "clusterName" => "test-cluster", "externalDNSName" => "dns", "keyName" => "key", "region" => "west", "availabilityZone" => "zone", "kmsKeyArn" => "arn" }
+    system "kube-aws", "init", "--cluster-name", "test-cluster", "--external-dns-name", "dns", "--region", "west", "--availability-zone", "zone", "--key-name", "key", "--kms-key-arn", "arn"
+    installed_version = shell_output("#{bin}/kube-aws version 2>&1")
+    cluster_info = YAML.load(File.read(File.join(Dir.pwd, "cluster.yaml")))
+
+    assert_equal cluster, cluster_info
```
Comment (jskswamy) [inline Formula/kube-aws.rb]: Added assertion to test the generated cluster.yml file

File: Formula/kube-aws.rb [hunk 9]
```diff
@@ -11,10 +11,10 @@ class KubeAws < Formula
...
   def install
     executable = "bin/kube-aws"
-    ENV["GOPATH_VENDOR"]="#{Dir.pwd}/_gopath-vendor"
-    ENV["GOPATH_KUBE_AWS"]="#{Dir.pwd}/_gopath-kube-aws"
-    ENV["KUBE_AWS_DIR"]="#{ENV["GOPATH_KUBE_AWS"]}/src/github.com/coreos/coreos-kubernetes/multi-node/aws"
-    ENV["GOPATH"]="#{ENV["GOPATH_VENDOR"]}:#{ENV["GOPATH_KUBE_AWS"]}"
+    ENV["GOPATH_VENDOR"] = "#{Dir.pwd}/_gopath-vendor"
+    ENV["GOPATH_KUBE_AWS"] = "#{Dir.pwd}/_gopath-kube-aws"
+    ENV["KUBE_AWS_DIR"] = "#{ENV["GOPATH_KUBE_AWS"]}/src/github.com/coreos/coreos-kubernetes/multi-node/aws"
+    ENV["GOPATH"] = "#{ENV["GOPATH_VENDOR"]}:#{ENV["GOPATH_KUBE_AWS"]}"
```
Comment (jskswamy) [inline Formula/kube-aws.rb]: Added space between operator for better readability

File: Formula/kube-aws.rb [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+  homepage "https://github.com/coreos/coreos-kubernetes/tree/master/multi-node/aws"
+  url "https://github.com/coreos/coreos-kubernetes/archive/v0.8.1.tar.gz"
+  sha256 "fe65c3fb72af7aa47dad07b8428eace077314edd3c129dba1fea5d3cccfaf094"
+  head "https://github.com/coreos/coreos-kubernetes.git"
+
+  depends_on "go" => :build
+
+  def install
+    executable = "bin/kube-aws"
+    ENV["GOPATH_VENDOR"] = "#{Dir.pwd}/_gopath-vendor"
```
Comment (dunn) [inline Formula/kube-aws.rb]: You can use `buildpath` instead of `Dir.pwd`, here and in the next line.

File: Formula/kube-aws.rb [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+    ENV["GOPATH_KUBE_AWS"] = "#{Dir.pwd}/_gopath-kube-aws"
+    ENV["KUBE_AWS_DIR"] = "#{ENV["GOPATH_KUBE_AWS"]}/src/github.com/coreos/coreos-kubernetes/multi-node/aws"
+    ENV["GOPATH"] = "#{ENV["GOPATH_VENDOR"]}:#{ENV["GOPATH_KUBE_AWS"]}"
+
+    Dir.chdir "#{Dir.pwd}/multi-node/aws/"
+
+    rm_rf [ENV["GOPATH_VENDOR"], ENV["GOPATH_KUBE_AWS"]]
```
Comment (dunn) [inline Formula/kube-aws.rb]: Why are we deleting these then recreating them below?  What do they contain originally?

File: Formula/kube-aws.rb [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+
+    mkdir_p ENV["GOPATH_VENDOR"]
+    mkdir_p File.dirname(ENV["KUBE_AWS_DIR"])
+
+    ln "#{Dir.pwd}/vendor", "#{ENV["GOPATH_VENDOR"]}/src"
```
Comment (dunn) [inline Formula/kube-aws.rb]: Use `ln_s` or `ln_sf` here and below.

File: Formula/kube-aws.rb [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+    ln Dir.pwd.to_s, (ENV["KUBE_AWS_DIR"]).to_s
```
Comment (dunn) [inline Formula/kube-aws.rb]: `ENV["KUBE_AWS_DIR"]` already is a string, so it doesn't need to be converted.

File: Formula/kube-aws.rb [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+
+    system "go", "generate", "./pkg/config"
+    system "go", "build", "-ldflags", "-X github.com/coreos/coreos-kubernetes/multi-node/aws/pkg/cluster.VERSION=#{version}", "-a", "-tags", "netgo", "-installsuffix", "netgo", "-o", executable.to_s, "./cmd/kube-aws"
```
Comment (dunn) [inline Formula/kube-aws.rb]: Replace `executable.to_s` with `bin/"kube-aws"` and you can delete the `bin.install` line below.  Then you can remove the `executable` variable entirely.

Also, please add a line break or two here for readability.

File: Formula/kube-aws.rb [hunk 15]
```diff
@@ -0,0 +1,44 @@
...
+  depends_on "go" => :build
+
+  def install
+    executable = "bin/kube-aws"
+    ENV["GOPATH_VENDOR"] = "#{Dir.pwd}/_gopath-vendor"
+    ENV["GOPATH_KUBE_AWS"] = "#{Dir.pwd}/_gopath-kube-aws"
+    ENV["KUBE_AWS_DIR"] = "#{ENV["GOPATH_KUBE_AWS"]}/src/github.com/coreos/coreos-kubernetes/multi-node/aws"
+    ENV["GOPATH"] = "#{ENV["GOPATH_VENDOR"]}:#{ENV["GOPATH_KUBE_AWS"]}"
+
+    Dir.chdir "#{Dir.pwd}/multi-node/aws/"
```
Comment (dunn) [inline Formula/kube-aws.rb]: Replace this with a `cd buildpath/"multi-node/aws" do` block.

**Review by dunn — CHANGES_REQUESTED** [review]
Sorry for the delay, but your formula is a bit complicated (probably more complicated than necessary), so I didn't get another chance to look through it until now.

File: Formula/kube-aws.rb [hunk 16]
```diff
@@ -11,25 +11,23 @@ class KubeAws < Formula
 
   def install
     executable = "bin/kube-aws"
-    ENV["GOPATH_VENDOR"] = "#{Dir.pwd}/_gopath-vendor"
-    ENV["GOPATH_KUBE_AWS"] = "#{Dir.pwd}/_gopath-kube-aws"
+    ENV["GOPATH_VENDOR"] = "#{buildpath}/_gopath-vendor"
```
Comment (jskswamy) [inline Formula/kube-aws.rb]: Using buildpath instead of Dir.pwd

File: Formula/kube-aws.rb [hunk 17]
```diff
@@ -11,25 +11,23 @@ class KubeAws < Formula
...
+    ENV["GOPATH_KUBE_AWS"] = "#{buildpath}/_gopath-kube-aws"
     ENV["KUBE_AWS_DIR"] = "#{ENV["GOPATH_KUBE_AWS"]}/src/github.com/coreos/coreos-kubernetes/multi-node/aws"
     ENV["GOPATH"] = "#{ENV["GOPATH_VENDOR"]}:#{ENV["GOPATH_KUBE_AWS"]}"
 
-    Dir.chdir "#{Dir.pwd}/multi-node/aws/"
```
Comment (jskswamy) [inline Formula/kube-aws.rb]: Using cd with block instead if Dir.chdir

File: Formula/kube-aws.rb [hunk 18]
```diff
@@ -11,25 +11,23 @@ class KubeAws < Formula
...
+    cd "#{buildpath}/multi-node/aws/" do
+      mkdir_p ENV["GOPATH_VENDOR"]
+      mkdir_p File.dirname(ENV["KUBE_AWS_DIR"])
 
-    rm_rf [ENV["GOPATH_VENDOR"], ENV["GOPATH_KUBE_AWS"]]
```
Comment (jskswamy) [inline Formula/kube-aws.rb]: Removed unnecessary code block

File: Formula/kube-aws.rb [hunk 19]
```diff
@@ -11,25 +11,23 @@ class KubeAws < Formula
...
+      system "go", "build", "-ldflags", "-X github.com/coreos/coreos-kubernetes/multi-node/aws/pkg/cluster.VERSION=#{version}", "-a", "-tags", "netgo", "-installsuffix", "netgo", "-o", executable, "./cmd/kube-aws"
 
-    ln "#{Dir.pwd}/vendor", "#{ENV["GOPATH_VENDOR"]}/src"
-    ln Dir.pwd.to_s, (ENV["KUBE_AWS_DIR"]).to_s
-
-    system "go", "generate", "./pkg/config"
-    system "go", "build", "-ldflags", "-X github.com/coreos/coreos-kubernetes/multi-node/aws/pkg/cluster.VERSION=#{version}", "-a", "-tags", "netgo", "-installsuffix", "netgo", "-o", executable.to_s, "./cmd/kube-aws"
-
-    bin.install "#{Dir.pwd}/#{executable}" => "kube-aws"
+      bin.install executable => "kube-aws"
```
Comment (jskswamy) [inline Formula/kube-aws.rb]: This is still required to avoid `Error: Empty installation`

Comment (dunn) [inline Formula/kube-aws.rb]: See https://github.com/Homebrew/homebrew-core/pull/4797#discussion_r79963070; you should be able to build directly into `bin`.

**Comment by MikeMcQuaid** [comment]
@jskswamy Any news on this?

**Comment by apjanke** [comment]
Looks like this PR has several commits, and they're not interacting with GitHub's new "code review" mechanism well: when viewed on github.com, it doesn't look like prior change requests have been addressed.

Would you mind squashing this to a single commit and rebasing on `master`? That might tidy things up here.

**Comment by MikeMcQuaid** [comment]
This was merged in https://github.com/Homebrew/homebrew-core/pull/5122.

**Pull request closed without merging**